src/mapgen/mapgen.cpp

```diff
@@ -370,19 +370,25 @@ inline bool Mapgen::isLiquidHorizontallyFlowable(u32 vi, v3s16 em)
 void Mapgen::updateLiquid(UniqueQueue<v3s16> *trans_liquid, v3s16 nmin, v3s16 nmax)
 {
 	bool isignored, isliquid, wasignored, wasliquid, waschecked, waspushed;
+	content_t was_n;
 	const v3s16 &em  = vm->m_area.getExtent();
 
+	isignored = wasignored = true;
+	isliquid = wasliquid = false;
+	was_n = CONTENT_IGNORE;
+
 	for (s16 z = nmin.Z + 1; z <= nmax.Z - 1; z++)
 	for (s16 x = nmin.X + 1; x <= nmax.X - 1; x++) {
-		wasignored = true;
-		wasliquid = false;
```

Comment:
Why don't you have to reset these at each node column?

Reply:
good point, I do